torch/utils/data/datapipes/utils/common.py

```diff
@@ -157,13 +157,59 @@ class StreamWrapper:
     DataPipe operation like `FileOpener`. StreamWrapper would guarantee
     the wrapped file handler is closed when it's out of scope.
     '''
-    def __init__(self, file_obj):
+    session_streams: Dict[Any, int] = {}
```

Comment:
We can use a `set` in this case.
Do you think we might also register a function to `atexit` to make sure all `session_streams` are closed?

Reply:
Oddly it doesn't allow to put self into Set. Switching back to Dict.